deepchem/models/torch_models/tests/test_grover_layers.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+
+try:
+    import torch
+except ModuleNotFoundError:
+    pass
+
+
+@pytest.mark.torch
+@pytest.mark.parametrize('dynamic_depth', ['none', 'uniform'])
+@pytest.mark.parametrize('atom_messages', [False, True])
+def testGroverMPNEncoder(dynamic_depth, atom_messages):
+    from deepchem.models.torch_models.grover_layers import GroverMPNEncoder
+    f_atoms = torch.randn(4, 151)
+    f_bonds = torch.randn(5, 165)
+    a2b = torch.Tensor([[0, 0], [2, 0], [4, 0], [1, 3]]).type(torch.int32)
+    b2a = torch.Tensor([0, 1, 3, 2, 3]).type(torch.long)
+    b2revb = torch.Tensor([0, 2, 1, 4, 3]).type(torch.long)
+    a2a = torch.Tensor([[0, 0], [3, 0], [3, 0], [1, 2]]).type(torch.int32)
+
+    # TODO Write tests for undirected = True case, currently fails. for this case, we have
+    # to generate inputs (a2b, b2a, b2revb) for undirected graph (mol2graph returns features
+    # for directed graphs)
```

Comment:
What is mol2graph?

Reply:
That was a note to self, based on original implementation of grover. I have removed it since it is not relevent here anymore.